.github/workflows/structured-logging-schema-check.yml

```diff
@@ -69,10 +63,3 @@ jobs:
       # we actually care if these pass, because the normal test run doesn't usually include many json log outputs
       - name: Run integration tests
         run: tox -e integration -- -nauto
-
-      # apply our schema tests to every log event from the previous step
-      # skips any output that isn't valid json
-      - uses: actions-rs/cargo@v1
-        with:
-          command: run
-          args: --manifest-path test/interop/log_parsing/Cargo.toml
```

Comment:
I removed the rust piece because a lot of it would have to be refactored and in my opinion it doesn't help that much. If someone disagrees or wants to do that re-implementation work, that's fine. I would personally vote for doing that in Python, so it can more easily be run locally.

The test as it is now does check that everything can be serialized and did catch a case where merging main had removed some updates to a fire_event call.

There's another ticket to implement log_format of protobuf and round trip the formatted logs.

Reply:
@nathaniel-may do you remember why this was written in Rust?

Reply:
We're consuming the logs from an "outside source" so it could be written in anything. I chose Rust because you can generate most of the code we'd be using here from a type definition. Plus Python's serialization-deserialization story is pretty abysmal so I didn't consider that a good option.

Reply:
So the way I see it, we only have two ways to verify that our structured logs conform to anything even remotely reasonable for our users to consume: 

1. mypy in [base_types.py](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/base_types.py) but mypy is limited on what it can check because it has to be defined at the type level. (e.g. - it can't catch positive vs negative integers)
1. These interop tests written in Rust. There are lots of values we can't check at the type level so I think good we check values since it's a programmatic interface.

I will concede that I didn't add very many value checks so @gshank is correct to call out it's limited usefulness in its current state. The original idea is that we would add more to these tests over time which we haven't really done. If Rust isn't the right choice for the team I'm perfectly fine to see this rewritten.

Reply:
Will open a ticket to look at restoring some of this testing.